Microsoft.Toolkit.Uwp.UI.Controls/Carousel/Carousel.cs

```diff
@@ -0,0 +1,397 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using Microsoft.Toolkit.Uwp.UI.Animations;
+using Windows.ApplicationModel.Core;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Control the user to cycle through a collection of items
+    /// </summary>
+    [TemplatePart(Name = PartCarouselPanel, Type = typeof(CarouselPanel))]
+    public sealed class Carousel : ContentControl
+    {
+        private const string PartCarouselPanel = "TPanel";
+        private const double _scrollRate = 100;
+        private int _originalIndex;
+        private CarouselPanel _carouselPanel;
+        private DataTemplate _itemTemplate;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Carousel"/> class.
+        /// </summary>
+        public Carousel()
+        {
+            this.DefaultStyleKey = typeof(Carousel);
```

Comment:
this is defendant, you can remove it as this the guidelines for the toolkit as you know :) also please remove all this in rest of classes